lib/gtfs/data.ex

```diff
@@ -83,6 +83,50 @@ defmodule Gtfs.Data do
     Block.get(blocks, block_id, service_id)
   end
 
+  @spec active_blocks(t(), Util.Time.timestamp(), Util.Time.timestamp()) :: %{
+          Route.id() => [Block.id()]
+        }
+  def active_blocks(%__MODULE__{trips: trips, calendar: calendar}, start_time, end_time) do
+    # We might cover multiple service dates
+    # If we start before 6am, check trips on the previous day
+    # It's okay if this date range is a little too broad, since we check trips by time on those dates later
+    first_possible_service_date =
+      start_time
+      |> (fn time -> time - 6 * 60 * 60 end).()
+      |> Util.Time.date_for_timestamp()
+
+    last_possible_service_date =
+      end_time
+      |> Util.Time.date_for_timestamp()
+
+    dates = Enum.to_list(Date.range(first_possible_service_date, last_possible_service_date))
```

Comment:
I don't like using pipes when neither of the arguments feels primary. In this case, I think splitting the start and end of the range onto different lines would be worse for readability than the nested parentheses. And since the dates are the second argument of the function call on the next line, that line can't be incorporated into a longer pipeline.

Reply:
Another option is an intermediate variable—`date_range` maybe. That also helps with the "hard to read nested function calls" problem.